apps/astarte_data_updater_plant/test/astarte_data_updater_plant/data_updater/time_based_actions_test.exs

```diff
@@ -0,0 +1,1016 @@
+#
+# This file is part of Astarte.
+#
+# Copyright 2025 SECO Mind Srl
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+defmodule Astarte.DataUpdaterPlant.TimeBasedActionsTest do
+  use ExUnit.Case, async: true
+  import Mox
+
+  alias Astarte.Core.Device
+  alias Astarte.Core.Triggers.SimpleTriggersProtobuf.AMQPTriggerTarget
+  alias Astarte.Core.Triggers.SimpleTriggersProtobuf.DataTrigger
+  alias Astarte.Core.Triggers.SimpleTriggersProtobuf.DeviceTrigger
+  alias Astarte.Core.Triggers.SimpleTriggersProtobuf.SimpleTriggerContainer
+  alias Astarte.Core.Triggers.SimpleTriggersProtobuf.TriggerTargetContainer
+  alias Astarte.DataUpdaterPlant.AMQPTestHelper
+  alias Astarte.DataUpdaterPlant.DataUpdater
+  alias Astarte.DataUpdaterPlant.TimeBasedActions
+  alias Astarte.Helpers.Database
+
+  @timestamp_us_x_10 Database.make_timestamp("2025-05-14T14:00:32+00:00")
+  @timestamp2_us_x_10 Database.make_timestamp("2025-05-14T14:10:32+00:00")
+  @interface_lifespan_decimicroseconds 60 * 10 * 1000 * 10000
+
+  setup_all do
+    realm = "autotestrealm#{System.unique_integer([:positive])}"
+    Database.setup!(realm)
+    on_exit(fn -> Database.teardown!(realm) end)
+    {:ok, realm: realm}
+  end
+
+  setup %{realm: realm} do
+    device_id = Database.random_device_id()
+    encoded_device_id = Device.encode_device_id(device_id)
+    {:ok, device_id: device_id, encoded_device_id: encoded_device_id, realm: realm}
+  end
```

Comment:
You can `use Astarte.Cases.Data, async: true` so that these kinds of things are already taken care of